src/sounds.cpp

```diff
@@ -388,16 +393,790 @@ std::pair<std::vector<tripoint>, std::vector<tripoint>> sounds::get_monster_soun
     return { sound_locations, cluster_centroids };
 }
 
-std::string sounds::sound_at( const tripoint &location )
-{
+std::string sounds::sound_at( const tripoint &location ) {
     auto this_sound = sound_markers.find( location );
     if( this_sound == sound_markers.end() ) {
         return std::string();
     }
-
     if( !this_sound->second.description.empty() ) {
         return this_sound->second.description;
     }
+    return _( "a sound" );
+}
+
+//sfx::
+void sfx::load_sound_effects( JsonObject &jsobj ) {
+    set_group_channels( 2, 9, 1 );
+    set_group_channels( 0, 1, 2 );
+    set_group_channels( 11, 14, 3 );
+    set_group_channels( 15, 17, 4 );
+    int index = 0;
+    std::string file;
+    sound_effect new_sound_effect;
+    new_sound_effect.id = jsobj.get_string( "id" );
+    new_sound_effect.volume = jsobj.get_int( "volume" );
+    new_sound_effect.variant = jsobj.get_string( "variant" );
+    JsonArray jsarr = jsobj.get_array( "files" );
+    while( jsarr.has_more() ) {
+        new_sound_effect.files.push_back( jsarr.next_string().c_str() );
+        file = new_sound_effect.files[index];
```

Comment:
`.c_str()` is completely superfluous, `next_string` already returns a perfectly fine string, which is what `push_back` requires.